docs/glossary.rst

```diff
@@ -0,0 +1,12 @@
+*************
+Term Glossary
+*************
+
+.. glossary::
+
+   Unit_like
+      Must be an `~astropy.units.UnitBase` object or a string parseable by
```

Comment:
Maybe
```
Must be an `~astropy.units.UnitBase` instance or a string or other instance
parseable by `~astropy.units.Unit`.
```
Partially for hyper-correctness: logarithmic units are not `UnitBase` instances... And we may end up deciding to interpret units from other unit packages.

Reply:
https://github.com/astropy/astropy/pull/11118/commits/72ba5228d00abd0c7d48c5cce588e9ac034a0991 implements this change